src/main/java/io/airlift/compress/zstd/FiniteStateEntropy.java

```diff
@@ -110,6 +111,7 @@ public int decompress(final Object inputBase, final long inputAddress, final lon
         }
 
         while (true) {
+            verify(output <= outputLimit - 2, input, "Output buffer is too small");
```

Comment:
I realize we use the same message in other places, but the actual problem is that something in the input is corrupt (especially since the output buffer is a constant size). Thus, it's rather misleading for the end user.

Reply:
Yeah, unfortunately, this is one of those cases where the real cause depends on context. From the point of view of this method, the output buffer *is* too small. From the point of view of the caller, the only way that error can happen is if the input is corrupted.